From e4a68fafc5dc5e3af68fc4cd765e60484bc890e7 Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Fri, 16 May 2025 13:40:17 +0200
Subject: [PATCH] vim: fix error E1187

Fix error E1187 for vim, where defaults.vim was not found
---
 utils/vim/Makefile                            | 19 ++++++++++---------
 utils/vim/files/{vimrc => defaults.vim}       |  0
 .../files/{vimrc.full => defaults.vim.full}   |  0
 3 files changed, 10 insertions(+), 9 deletions(-)
 rename utils/vim/files/{vimrc => defaults.vim} (100%)
 rename utils/vim/files/{vimrc.full => defaults.vim.full} (100%)

diff --git a/utils/vim/Makefile b/utils/vim/Makefile
index 125acbf99..aef74c1ce 100644
--- a/utils/vim/Makefile
+++ b/utils/vim/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=vim
 PKG_VERSION:=9.0
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 VIMVER:=90
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
@@ -73,12 +73,12 @@ define Package/xxd
 endef
 
 define Package/vim-full/conffiles
-/usr/share/vim/vimrc
+/usr/share/vim/defaults.vim
 /root/.vimrc
 endef
 
 define Package/vim/conffiles
-/usr/share/vim/vimrc
+/usr/share/vim/defaults.vim
 /root/.vimrc
 endef
 
@@ -193,6 +193,7 @@ define Build/Compile/vim-runtime
 	$(MAKE) -C $(PKG_BUILD_DIR)/src DESTDIR="$(PKG_INSTALL_DIR)" installrtbase
 	(cd $(PKG_INSTALL_DIR) && tar -cf $(PKG_BUILD_DIR)/docs.tar ./usr/share/vim/vim$(VIMVER)/doc)
 	rm -rf $(PKG_INSTALL_DIR)/usr/share/vim/vim$(VIMVER)/doc
+	rm -rf $(PKG_INSTALL_DIR)/usr/share/vim/vim$(VIMVER)/defaults.vim
 	rm -rf $(PKG_INSTALL_DIR)/usr/man
 endef
 
@@ -207,26 +208,26 @@ endef
 define Package/vim/install
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/vim_tiny $(1)/usr/bin/vim
-	$(INSTALL_DIR) $(1)/usr/share/vim
-	$(INSTALL_CONF) ./files/vimrc $(1)/usr/share/vim/
+	$(INSTALL_DIR) $(1)/usr/share/vim/vim$(VIMVER)
+	$(INSTALL_CONF) ./files/defaults.vim $(1)/usr/share/vim/vim$(VIMVER)
 endef
 
 define Package/vim-full/install
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/vim_normal $(1)/usr/bin/vim
-	$(INSTALL_DIR) $(1)/usr/share/vim
+	$(INSTALL_DIR) $(1)/usr/share/vim/vim$(VIMVER)
 	$(LN) vim $(1)/usr/bin/vimdiff
-	$(INSTALL_CONF) ./files/vimrc.full $(1)/usr/share/vim/vimrc
+	$(INSTALL_CONF) ./files/defaults.vim.full $(1)/usr/share/vim/vim$(VIMVER)/defaults.vim
 endef
 
 
 define Package/vim-fuller/install
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/vim_big $(1)/usr/bin/vim
-	$(INSTALL_DIR) $(1)/usr/share/vim
+	$(INSTALL_DIR) $(1)/usr/share/vim/vim$(VIMVER)
 	$(LN) vim $(1)/usr/bin/vimdiff
 	$(CP) $(PKG_INSTALL_DIR)/usr/share/vim/vim$(VIMVER) $(1)/usr/share/vim
-	$(INSTALL_CONF) ./files/vimrc.full $(1)/usr/share/vim/vimrc
+	$(INSTALL_CONF) ./files/defaults.vim.full $(1)/usr/share/vim/vim$(VIMVER)/defaults.vim
 endef
 
 
diff --git a/utils/vim/files/vimrc b/utils/vim/files/defaults.vim
similarity index 100%
rename from utils/vim/files/vimrc
rename to utils/vim/files/defaults.vim
diff --git a/utils/vim/files/vimrc.full b/utils/vim/files/defaults.vim.full
similarity index 100%
rename from utils/vim/files/vimrc.full
rename to utils/vim/files/defaults.vim.full
-- 
2.49.0

